lib/grape-swagger/endpoint.rb

```diff
@@ -235,6 +235,11 @@ def parse_request_params(required)
       end
     end
 
+    def param_type_is_array?(param_type)
+      return false unless param_type
+      param_type == 'Array' || param_type =~ /\[\w+\]\z/
+    end
```

Comment:
You're parsing this twice, perpetuating the mess we have here and adding to it :) Refactor `param_type_is_array` into `parse_param_type` or something like that and return the array key. Nuke the whole `possible_key = param.first.to_s.gsub('[', '[][')` as well as part of that.